### ISSUE TITLE
Summary<TEndpoint, TRequest> throws a runtime error

### ISSUE THREAD

**Issue by OFark** [dj-nitehawk/FastEndpoints#154]
.Net core 6 API and I tried using the 
`MySummary : Summary<TEndpoint, TRequest>` class for creating summaries. 

It compiles, but the runtime throws this error
```
Unhandled exception. System.ArgumentNullException: Value cannot be null. (Parameter 'key')
   at System.Collections.Generic.Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior)
   at FastEndpoints.EndpointData.BuildEndpointDefinitions(IServiceCollection services, IEnumerable`1 assemblies, ConfigurationManager config)
   at FastEndpoints.EndpointData.<>c__DisplayClass6_0.<.ctor>b__0()
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
   at System.Lazy`1.ExecutionAndPublication(LazyHelper executionAndPublication, Boolean useDefaultConstructor)
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.get_Value()
   at FastEndpoints.EndpointData..ctor(IServiceCollection services, IEnumerable`1 assemblies, ConfigurationManager config)
   at FastEndpoints.MainExtensions.AddFastEndpoints(IServiceCollection services, IEnumerable`1 endpointAssmeblies, ConfigurationManager config)
  at Program.<Main>$(String[] args) in - "builder.Services.AddFastEndpoints();"
```

If I remove the second generic, TRequest, it works just fine, but I no longer have any reference to my request.

**Comment by dj-nitehawk** [comment]
i believe this is fixed in `v4.3.0-beta11`
pls try that version: https://www.nuget.org/packages/FastEndpoints/4.3.0-beta11

**Comment by OFark** [comment]
I see you pushed 4.3.0 over the weekend. This fixed the problem for me and it has allowed me to use this method to base class the summaries to solve my other problem with the XML summary base class issue. #153

**Comment by OFark** [comment]
Have you tried hosting in IIS. It's the only way I can replicate the issue. It seems after IIS has terminated the app due to inactivity any requests to the swagger json results in a 500 

I had Serilog dump the exception to Elasticsearch:
```
{
  "_index": "ianua-logs-development-2022-08",
  "_type": "_doc",
  "_id": "x4YSX4IBFRXkwPqd-nA8",
  "_version": 1,
  "_score": 1,
  "_ignored": [
    "exceptions.StackTraceString.keyword"
  ],
  "_source": {
    "@timestamp": "2022-08-02T16:00:55.2673911+01:00",
    "level": "Error",
    "messageTemplate": "An unhandled exception has occurred while executing the request.",
    "message": "An unhandled exception has occurred while executing the request.",
    "exceptions": [
      {
        "Depth": 0,
        "ClassName": "System.NullReferenceException",
        "Message": "Object reference not set to an instance of an object.",
        "Source": "FastEndpoints.Swagger",
        "StackTraceString": "   at FastEndpoints.Swagger.ValidationSchemaProcessor.Process(SchemaProcessorContext context)\r\n   at NJsonSchema.Generation.JsonSchemaGenerator.ApplySchemaProcessors(JsonSchema schema, ContextualType contextualType, JsonSchemaResolver schemaResolver)\r\n   at NJsonSchema.Generation.JsonSchemaGenerator.Generate[TSchemaType](TSchemaType schema, ContextualType contextualType, JsonSchemaResolver schemaResolver)\r\n   at NJsonSchema.Generation.JsonSchemaGenerator.Generate[TSchemaType](ContextualType contextualType, JsonSchemaResolver schemaResolver)\r\n   at NJsonSchema.Generation.JsonSchemaGenerator.GenerateWithReferenceAndNullability[TSchemaType](ContextualType contextualType, Boolean isNullable, JsonSchemaResolver schemaResolver, Action`2 transformation)\r\n   at NSwag.Generation.OpenApiSchemaGenerator.GenerateWithReferenceAndNullability[TSchemaType](ContextualType contextualType, Boolean isNullable, JsonSchemaResolver schemaResolver, Action`2 transformation)\r\n   at NJsonSchema.Generation.JsonSchemaGenerator.LoadPropertyOrField(JsonProperty jsonProperty, ContextualAccessorInfo accessorInfo, Type parentType, JsonSchema parentSchema, JsonSchemaResolver schemaResolver)\r\n   at NJsonSchema.Generation.JsonSchemaGenerator.GenerateProperties(Type type, JsonSchema schema, JsonSchemaResolver schemaResolver)\r\n   at NJsonSchema.Generation.JsonSchemaGenerator.GenerateObject(JsonSchema schema, JsonTypeDescription typeDescription, JsonSchemaResolver schemaResolver)\r\n   at NSwag.Generation.OpenApiSchemaGenerator.GenerateObject(JsonSchema schema, JsonTypeDescription typeDescription, JsonSchemaResolver schemaResolver)\r\n   at NJsonSchema.Generation.JsonSchemaGenerator.Generate[TSchemaType](TSchemaType schema, ContextualType contextualType, JsonSchemaResolver schemaResolver)\r\n   at NJsonSchema.Generation.JsonSchemaGenerator.Generate[TSchemaType](ContextualType contextualType, JsonSchemaResolver schemaResolver)\r\n   at NJsonSchema.Generation.JsonSchemaGenerator.GenerateWithReferenceAndNullability[TSchemaType](ContextualType contextualType, Boolean isNullable, JsonSchemaResolver schemaResolver, Action`2 transformation)\r\n   at NSwag.Generation.OpenApiSchemaGenerator.GenerateWithReferenceAndNullability[TSchemaType](ContextualType contextualType, Boolean isNullable, JsonSchemaResolver schemaResolver, Action`2 transformation)\r\n   at NSwag.Generation.AspNetCore.Processors.OperationResponseProcessor.Process(OperationProcessorContext operationProcessorContext)\r\n   at NSwag.Generation.AspNetCore.AspNetCoreOpenApiDocumentGenerator.RunOperationProcessors(OpenApiDocument document, ApiDescription apiDescription, Type controllerType, MethodInfo methodInfo, OpenApiOperationDescription operationDescription, List`1 allOperations, OpenApiDocumentGenerator swaggerGenerator, OpenApiSchemaResolver schemaResolver)\r\n   at NSwag.Generation.AspNetCore.AspNetCoreOpenApiDocumentGenerator.AddOperationDescriptionsToDocument(OpenApiDocument document, Type controllerType, List`1 operations, OpenApiDocumentGenerator swaggerGenerator, OpenApiSchemaResolver schemaResolver)\r\n   at NSwag.Generation.AspNetCore.AspNetCoreOpenApiDocumentGenerator.GenerateApiGroups(OpenApiDocument document, IGrouping`2[] apiGroups, OpenApiSchemaResolver schemaResolver)\r\n   at NSwag.Generation.AspNetCore.AspNetCoreOpenApiDocumentGenerator.GenerateAsync(ApiDescriptionGroupCollection apiDescriptionGroups)\r\n   at NSwag.AspNetCore.Middlewares.OpenApiDocumentMiddleware.GenerateDocumentAsync(HttpContext context)\r\n   at NSwag.AspNetCore.Middlewares.OpenApiDocumentMiddleware.GetDocumentAsync(HttpContext context)\r\n   at NSwag.AspNetCore.Middlewares.OpenApiDocumentMiddleware.Invoke(HttpContext context)\r\n   at NSwag.AspNetCore.Middlewares.SwaggerUiIndexMiddleware.Invoke(HttpContext context)\r\n   at NSwag.AspNetCore.Middlewares.RedirectToIndexMiddleware.Invoke(HttpContext context)\r\n   at Presentation.Middleware.APIKeyAuthenticator.InvokeAsync(HttpContext context, IAuthenticationService authenticationService, IOptionsSnapshot`1 options, ILogger logger) in D:\\a\\1\\s\\Presentation\\Middleware\\APIKeyAuthenticator.cs:line 55\r\n   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)\r\n   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)",
        "RemoteStackTraceString": null,
        "RemoteStackIndex": 0,
        "HResult": -2147467261,
        "HelpURL": null
      }
    ],
    "fields": {
      "EventId": {
        "Id": 1,
        "Name": "UnhandledException"
      },
      "SourceContext": "Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware",
      "RequestId": "800000fd-0006-d900-b63f-84710c7967bb",
      "RequestPath": "/swagger/Initial%20Release/swagger.json",
      "MachineName": "SWV082DEV1",
      "Environment": "Development"
    }
  },
  "fields": {
    "fields.RequestPath.keyword": [
      "/swagger/Initial%20Release/swagger.json"
    ],
    "exceptions.Source.keyword": [
      "FastEndpoints.Swagger"
    ],
    "fields.Environment.keyword": [
      "Development"
    ],
    "exceptions.StackTraceString": [
      "   at FastEndpoints.Swagger.ValidationSchemaProcessor.Process(SchemaProcessorContext context)\r\n   at NJsonSchema.Generation.JsonSchemaGenerator.ApplySchemaProcessors(JsonSchema schema, ContextualType contextualType, JsonSchemaResolver schemaResolver)\r\n   at NJsonSchema.Generation.JsonSchemaGenerator.Generate[TSchemaType](TSchemaType schema, ContextualType contextualType, JsonSchemaResolver schemaResolver)\r\n   at NJsonSchema.Generation.JsonSchemaGenerator.Generate[TSchemaType](ContextualType contextualType, JsonSchemaResolver schemaResolver)\r\n   at NJsonSchema.Generation.JsonSchemaGenerator.GenerateWithReferenceAndNullability[TSchemaType](ContextualType contextualType, Boolean isNullable, JsonSchemaResolver schemaResolver, Action`2 transformation)\r\n   at NSwag.Generation.OpenApiSchemaGenerator.GenerateWithReferenceAndNullability[TSchemaType](ContextualType contextualType, Boolean isNullable, JsonSchemaResolver schemaResolver, Action`2 transformation)\r\n   at NJsonSchema.Generation.JsonSchemaGenerator.LoadPropertyOrField(JsonProperty jsonProperty, ContextualAccessorInfo accessorInfo, Type parentType, JsonSchema parentSchema, JsonSchemaResolver schemaResolver)\r\n   at NJsonSchema.Generation.JsonSchemaGenerator.GenerateProperties(Type type, JsonSchema schema, JsonSchemaResolver schemaResolver)\r\n   at NJsonSchema.Generation.JsonSchemaGenerator.GenerateObject(JsonSchema schema, JsonTypeDescription typeDescription, JsonSchemaResolver schemaResolver)\r\n   at NSwag.Generation.OpenApiSchemaGenerator.GenerateObject(JsonSchema schema, JsonTypeDescription typeDescription, JsonSchemaResolver schemaResolver)\r\n   at NJsonSchema.Generation.JsonSchemaGenerator.Generate[TSchemaType](TSchemaType schema, ContextualType contextualType, JsonSchemaResolver schemaResolver)\r\n   at NJsonSchema.Generation.JsonSchemaGenerator.Generate[TSchemaType](ContextualType contextualType, JsonSchemaResolver schemaResolver)\r\n   at NJsonSchema.Generation.JsonSchemaGenerator.GenerateWithReferenceAndNullability[TSchemaType](ContextualType contextualType, Boolean isNullable, JsonSchemaResolver schemaResolver, Action`2 transformation)\r\n   at NSwag.Generation.OpenApiSchemaGenerator.GenerateWithReferenceAndNullability[TSchemaType](ContextualType contextualType, Boolean isNullable, JsonSchemaResolver schemaResolver, Action`2 transformation)\r\n   at NSwag.Generation.AspNetCore.Processors.OperationResponseProcessor.Process(OperationProcessorContext operationProcessorContext)\r\n   at NSwag.Generation.AspNetCore.AspNetCoreOpenApiDocumentGenerator.RunOperationProcessors(OpenApiDocument document, ApiDescription apiDescription, Type controllerType, MethodInfo methodInfo, OpenApiOperationDescription operationDescription, List`1 allOperations, OpenApiDocumentGenerator swaggerGenerator, OpenApiSchemaResolver schemaResolver)\r\n   at NSwag.Generation.AspNetCore.AspNetCoreOpenApiDocumentGenerator.AddOperationDescriptionsToDocument(OpenApiDocument document, Type controllerType, List`1 operations, OpenApiDocumentGenerator swaggerGenerator, OpenApiSchemaResolver schemaResolver)\r\n   at NSwag.Generation.AspNetCore.AspNetCoreOpenApiDocumentGenerator.GenerateApiGroups(OpenApiDocument document, IGrouping`2[] apiGroups, OpenApiSchemaResolver schemaResolver)\r\n   at NSwag.Generation.AspNetCore.AspNetCoreOpenApiDocumentGenerator.GenerateAsync(ApiDescriptionGroupCollection apiDescriptionGroups)\r\n   at NSwag.AspNetCore.Middlewares.OpenApiDocumentMiddleware.GenerateDocumentAsync(HttpContext context)\r\n   at NSwag.AspNetCore.Middlewares.OpenApiDocumentMiddleware.GetDocumentAsync(HttpContext context)\r\n   at NSwag.AspNetCore.Middlewares.OpenApiDocumentMiddleware.Invoke(HttpContext context)\r\n   at NSwag.AspNetCore.Middlewares.SwaggerUiIndexMiddleware.Invoke(HttpContext context)\r\n   at NSwag.AspNetCore.Middlewares.RedirectToIndexMiddleware.Invoke(HttpContext context)\r\n   at Presentation.Middleware.APIKeyAuthenticator.InvokeAsync(HttpContext context, IAuthenticationService authenticationService, IOptionsSnapshot`1 options, ILogger logger) in D:\\a\\1\\s\\Presentation\\Middleware\\APIKeyAuthenticator.cs:line 55\r\n   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)\r\n   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)"
    ],
    "exceptions.RemoteStackIndex": [
      0
    ],
    "exceptions.Message": [
      "Object reference not set to an instance of an object."
    ],
    "fields.SourceContext": [
      "Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware"
    ],
    "fields.EventId.Name.keyword": [
      "UnhandledException"
    ],
    "fields.EventId.Name": [
      "UnhandledException"
    ],
    "exceptions.HResult": [
      -2147467261
    ],
    "fields.EventId.Id": [
      1
    ],
    "exceptions.ClassName.keyword": [
      "System.NullReferenceException"
    ],
    "exceptions.ClassName": [
      "System.NullReferenceException"
    ],
    "messageTemplate.keyword": [
      "An unhandled exception has occurred while executing the request."
    ],
    "fields.SourceContext.keyword": [
      "Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware"
    ],
    "fields.Environment": [
      "Development"
    ],
    "exceptions.Message.keyword": [
      "Object reference not set to an instance of an object."
    ],
    "exceptions.Depth": [
      0
    ],
    "level": [
      "Error"
    ],
    "fields.MachineName": [
      "SWV082DEV1"
    ],
    "message": [
      "An unhandled exception has occurred while executing the request."
    ],
    "fields.RequestId.keyword": [
      "800000fd-0006-d900-b63f-84710c7967bb"
    ],
    "@timestamp": [
      "2022-08-02T15:00:55.267Z"
    ],
    "fields.RequestId": [
      "800000fd-0006-d900-b63f-84710c7967bb"
    ],
    "level.keyword": [
      "Error"
    ],
    "fields.RequestPath": [
      "/swagger/Initial%20Release/swagger.json"
    ],
    "message.keyword": [
      "An unhandled exception has occurred while executing the request."
    ],
    "exceptions.Source": [
      "FastEndpoints.Swagger"
    ],
    "messageTemplate": [
      "An unhandled exception has occurred while executing the request."
    ],
    "fields.MachineName.keyword": [
      "SWV082DEV1"
    ]
  },
  "ignored_field_values": {
    "exceptions.StackTraceString.keyword": [
      "   at FastEndpoints.Swagger.ValidationSchemaProcessor.Process(SchemaProcessorContext context)\r\n   at NJsonSchema.Generation.JsonSchemaGenerator.ApplySchemaProcessors(JsonSchema schema, ContextualType contextualType, JsonSchemaResolver schemaResolver)\r\n   at NJsonSchema.Generation.JsonSchemaGenerator.Generate[TSchemaType](TSchemaType schema, ContextualType contextualType, JsonSchemaResolver schemaResolver)\r\n   at NJsonSchema.Generation.JsonSchemaGenerator.Generate[TSchemaType](ContextualType contextualType, JsonSchemaResolver schemaResolver)\r\n   at NJsonSchema.Generation.JsonSchemaGenerator.GenerateWithReferenceAndNullability[TSchemaType](ContextualType contextualType, Boolean isNullable, JsonSchemaResolver schemaResolver, Action`2 transformation)\r\n   at NSwag.Generation.OpenApiSchemaGenerator.GenerateWithReferenceAndNullability[TSchemaType](ContextualType contextualType, Boolean isNullable, JsonSchemaResolver schemaResolver, Action`2 transformation)\r\n   at NJsonSchema.Generation.JsonSchemaGenerator.LoadPropertyOrField(JsonProperty jsonProperty, ContextualAccessorInfo accessorInfo, Type parentType, JsonSchema parentSchema, JsonSchemaResolver schemaResolver)\r\n   at NJsonSchema.Generation.JsonSchemaGenerator.GenerateProperties(Type type, JsonSchema schema, JsonSchemaResolver schemaResolver)\r\n   at NJsonSchema.Generation.JsonSchemaGenerator.GenerateObject(JsonSchema schema, JsonTypeDescription typeDescription, JsonSchemaResolver schemaResolver)\r\n   at NSwag.Generation.OpenApiSchemaGenerator.GenerateObject(JsonSchema schema, JsonTypeDescription typeDescription, JsonSchemaResolver schemaResolver)\r\n   at NJsonSchema.Generation.JsonSchemaGenerator.Generate[TSchemaType](TSchemaType schema, ContextualType contextualType, JsonSchemaResolver schemaResolver)\r\n   at NJsonSchema.Generation.JsonSchemaGenerator.Generate[TSchemaType](ContextualType contextualType, JsonSchemaResolver schemaResolver)\r\n   at NJsonSchema.Generation.JsonSchemaGenerator.GenerateWithReferenceAndNullability[TSchemaType](ContextualType contextualType, Boolean isNullable, JsonSchemaResolver schemaResolver, Action`2 transformation)\r\n   at NSwag.Generation.OpenApiSchemaGenerator.GenerateWithReferenceAndNullability[TSchemaType](ContextualType contextualType, Boolean isNullable, JsonSchemaResolver schemaResolver, Action`2 transformation)\r\n   at NSwag.Generation.AspNetCore.Processors.OperationResponseProcessor.Process(OperationProcessorContext operationProcessorContext)\r\n   at NSwag.Generation.AspNetCore.AspNetCoreOpenApiDocumentGenerator.RunOperationProcessors(OpenApiDocument document, ApiDescription apiDescription, Type controllerType, MethodInfo methodInfo, OpenApiOperationDescription operationDescription, List`1 allOperations, OpenApiDocumentGenerator swaggerGenerator, OpenApiSchemaResolver schemaResolver)\r\n   at NSwag.Generation.AspNetCore.AspNetCoreOpenApiDocumentGenerator.AddOperationDescriptionsToDocument(OpenApiDocument document, Type controllerType, List`1 operations, OpenApiDocumentGenerator swaggerGenerator, OpenApiSchemaResolver schemaResolver)\r\n   at NSwag.Generation.AspNetCore.AspNetCoreOpenApiDocumentGenerator.GenerateApiGroups(OpenApiDocument document, IGrouping`2[] apiGroups, OpenApiSchemaResolver schemaResolver)\r\n   at NSwag.Generation.AspNetCore.AspNetCoreOpenApiDocumentGenerator.GenerateAsync(ApiDescriptionGroupCollection apiDescriptionGroups)\r\n   at NSwag.AspNetCore.Middlewares.OpenApiDocumentMiddleware.GenerateDocumentAsync(HttpContext context)\r\n   at NSwag.AspNetCore.Middlewares.OpenApiDocumentMiddleware.GetDocumentAsync(HttpContext context)\r\n   at NSwag.AspNetCore.Middlewares.OpenApiDocumentMiddleware.Invoke(HttpContext context)\r\n   at NSwag.AspNetCore.Middlewares.SwaggerUiIndexMiddleware.Invoke(HttpContext context)\r\n   at NSwag.AspNetCore.Middlewares.RedirectToIndexMiddleware.Invoke(HttpContext context)\r\n   at Presentation.Middleware.APIKeyAuthenticator.InvokeAsync(HttpContext context, IAuthenticationService authenticationService, IOptionsSnapshot`1 options, ILogger logger) in D:\\a\\1\\s\\Presentation\\Middleware\\APIKeyAuthenticator.cs:line 55\r\n   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)\r\n   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)"
    ]
  }
}
```

**Comment by dj-nitehawk** [comment]
i believe this is the same thing as this one #173
we're trying to get to the bottom of it.
iis killing the process is something we haven't considered.
pls watch #173 for any new developments.
thanks for the log entry.

**Comment by dj-nitehawk** [comment]
if i remember IIS correctly, there's an option to SUSPEND after inactivity instead of TERMINATE right?
could you try setting it to suspend and see if the problem goes away?

**Comment by OFark** [comment]
I cannot get the Application Pool settings, regarding recycling, idling, termination style, or login as a user, to reproduce the issue, it does just seem to happen "randomly". I'll also check the event logs next time, but I have implemented 4.5.0-beta to test wjax's logging. So I'll send that to the #173 when I get a log.
